src/App.tsx

```diff
@@ -20,12 +22,16 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={(event) => {
+                  setQuery(event.target.value);
+                }}
               />
             </div>
           </div>
         </div>
 
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={moviesFromServer} query={query} />
```

Comment:
Instead of passing the query you need to create `visibleMovies` array. Check `readme` file.

Reply:
Re-implemented the task in a much simpler manner.